flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1206,8 +1206,12 @@ public Void retrievePayload(ResourceID resourceID) {
 	//  Resource Management
 	// ------------------------------------------------------------------------
 
-	protected int getNumberRequiredTaskManagerSlots() {
-		return slotManager.getNumberPendingTaskManagerSlots();
+	protected int getNumberRequiredTaskManagers() {
+		return getPendingWorkerNums().values().stream().reduce(0, Integer::sum);
```

Comment:
For the time being I would add a check state which ensures that the `WorkerResourceSpecs` can all be fulfilled by the container size we are using to start new containers/pods.

Reply:
IIUC, this is because the RM implementations are ignoring the `WorkerResourceSpec` passed to `startNewWorker`? Then would it be better to add this check in `startNewWorker`?

Reply:
If this is the entry point, then we can do the check there. The important bit is to maintain the invariant that we do not start containers for `WorkerResourceSpecs` which are larger than the container.